src/coreclr/vm/jitinterface.cpp

```diff
@@ -7868,6 +7868,38 @@ bool CEEInfo::haveSameMethodDefinition(
     return result;
 }
 
+CORINFO_CLASS_HANDLE CEEInfo::getTypeDefinition(CORINFO_CLASS_HANDLE type)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CORINFO_CLASS_HANDLE result = NULL;
+
+    JIT_TO_EE_TRANSITION_LEAF();
+
+    TypeHandle constructedHandle(type);
+    TypeHandle definitionHandle = ClassLoader::LoadTypeDefThrowing(
+        constructedHandle.GetModule(),
+        constructedHandle.GetMethodTable()->GetCl(),
+        ClassLoader::ReturnNullIfNotFound,
+        ClassLoader::PermitUninstDefOrRef,
+        /* tokenNotToLoad */ tdAllTypes);
+
+    result = CORINFO_CLASS_HANDLE(definitionHandle.AsPtr());
+
+    JIT_TO_EE_TRANSITION_LEAF();    
+
+    // We forbid loading new types here, and we expect the resulting types to
+    // always be available. That is because if we are trying to get the type
+    // definition of some 'Foo<Bar>' type, then 'Foo<>' must also be loaded.
```

Comment:
> we are trying to get the type definition of some 'Foo<Bar>' type, then 'Foo<>' must also be loaded.

This is not a safe assumption. We can load specific instantiation without loading the generic type definition.

Reply:
Oh, TIL, I didn't know that was possible at all. Thank you! 😄